src/org/labkey/test/tests/SampleTypeNameExpressionTest.java

```diff
@@ -180,6 +192,67 @@ public void testSimpleNameExpression()
         assertThat(names.get(2), startsWith("a-b.1." + batchRandomId + "."));
     }
 
+
+    // Issue 50924: LKSM: Importing samples using naming expression referencing parent inputs with # result in error
+    @Test
+    public void testDeriveFromCommentLikeParents()
+    {
+        final String sampleTypeName = "ParentNamesExprTest";
+
+        log("Verify import tsv would ignore lines starting with #");
+        String nameExpression = "${MaterialInputs/" + PARENT_SAMPLE_TYPE + "}-child";
+        String data = "MaterialInputs/" + PARENT_SAMPLE_TYPE + "\n";
+        data += PARENT_SAMPLE_01 + "\n";
+        data += PARENT_SAMPLE_01 + "," + PARENT_SAMPLE_02 + "," + PARENT_SAMPLE_03 + "\n";
+        // tsv lines starting with # should be ignored
+        data += PARENT_SAMPLE_03 + "\n";
+        data += PARENT_SAMPLE_03 + "," + PARENT_SAMPLE_02 + "\n";
```

Comment:
What if the sampleID is not the first field in the file?  That should succeed in creating the sample, right?

Reply:
Yes, there is a test case for that immediately below.

Reply:
I see the test for the parent input being not the first field, but I was wondering about when creating the sample itself.

Reply:
I see, I will add a test case for that. As long as the line doesn't start with #, it should work

Reply:
test added